tests/src/test/java/org/eclipse/hono/tests/mqtt/MqttPublishTestBase.java

```diff
@@ -911,11 +913,22 @@ protected void doTestUploadMessages(
                                 return oldValue != null ? null : promise; // remove mapping if oldValue is set
                             });
                         }
+                    } else {
+                        if (sendAttempt.failed()) {
+                            LOGGER.error("error sending message (success expected)l", sendAttempt.cause());;
+                            ctx.failNow(new IllegalStateException("error sending message (success expected)l"));
+                        }
```

Comment:
If sendAttempt never fail that why it is checked on case errorMessagesExpected is true (in other case)?
If it could fail  somehow - it is not expected in that errorMessagesExpected false case.- so, fail the test.